common/src/edgeai_pre_proc.cpp

```diff
@@ -88,32 +88,32 @@ int32_t preProc::getConfig(const string         &modelBasePath,
         LOG_ERROR("Preprocess configuration parameters missing.\n");
         status = -1;
     }
-    else if (session["input_mean"].IsNull())
```

Comment:
What is the "type" of session["input_mean"].  Is it boolean?  If not, then you should not use it alone in a conditional as per MISRAC.  The compiler think it is fine, but MISRAC is more stringent and it requires a more specific condition.  When I mentioned this to Shubham before, he added the .IsNull() as it returns boolean value.  If it is allowed that these are NULL, then what is the error that is being checked here?  That the object is equal to zero?  If that is the case, the condition should be (0 == session["input_mean"]).  Please confirm.

Reply:
I tried adding IsNull(), but it is not correct functionally according to the condition which is being checked here. Basically, an error should be thrown if that key does not exists in  YAML file. But IsNull() checks for the key and then judges whether the value is "null" or not there and this is not what we want to check here.

I looked at some documentation and code examples for YAML-CPP, everywhere they have used this way to check whether the key exists or not.

Even the official documentation have this reference:
https://github.com/jbeder/yaml-cpp/wiki/Tutorial

Querying for keys does not create them automatically (this makes handling optional map entries very easy)

YAML::Node node = YAML::Load("{name: Brewers, city: Milwaukee}");
if (node["name"]) {
  std::cout << node["name"].as<std::string>() << "\n";
}
if (node["mascot"]) {
  std::cout << node["mascot"].as<std::string>() << "\n";
}
assert(node.size() == 2); // the previous call didn't create a node

Reply:
Okay, thanks for checking.  In this case, when we run the MISRAC++ check, we may need to take an exception.  The waiver in this case will need to be against MISRAC++:2008 5-0-13.